src/components/Input.tsx

```diff
@@ -167,23 +170,108 @@ const InputBase = styled.input<
     }
   `}
 
+  ${({ theme, hasError }) =>
+    hasError &&
+    `
+    & + div ${NotchLeading},
+    & + div ${NotchTrailing},
+    & + div ${LabelNotchWrapper} {
+    border-color: ${theme.color["danger.500"]};
+    } 
+  `}
+
+  ${({ hasSuccess }) =>
+    hasSuccess &&
+    `
+    & + div ${NotchLeading},
+    & + div ${NotchTrailing},
+    & + div ${LabelNotchWrapper} {
+    border-color: #04C577;
```

Comment:
Please no hard coded colors. Maybe error and success colors can be passed as config ?